containerfixture/containerfixture.go

```diff
@@ -0,0 +1,119 @@
+package containerfixture
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"math/rand"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+//go:embed containerfixture.yaml
+var manifest string
+
+const Username = "admin"
+const Password = "Password"
+
+// Generates a random dynamic port number
+func GetRandomPort() string {
+	minPort := 49152
+	maxPort := 65535
+	rand.Seed(time.Now().UnixNano())
+
+	randomPort := fmt.Sprintf("%d", rand.Intn(maxPort-minPort)+minPort)
+	return randomPort
+}
+
+// Modify the manifest to allow running it in parallel as
+// many times as needed with dynamic values
+func modifyYaml(yaml string) map[string]string {
+	podName := fmt.Sprintf("test-pod-%s", GetRandomPort())
+	webPort := GetRandomPort()
+	apiPort := GetRandomPort()
+
+	yaml = strings.Replace(yaml, "name: \"test-pod\"", fmt.Sprintf("name: %s", podName), 1)
+	yaml = strings.Replace(yaml, "hostPort: 8000", fmt.Sprintf("hostPort: %s", webPort), 1)
+	yaml = strings.Replace(yaml, "hostPort: 8089", fmt.Sprintf("hostPort: %s", apiPort), 1)
+
+	return map[string]string{
+		"yaml":    yaml,
+		"podName": podName,
+		"webPort": webPort,
+		"apiPort": apiPort,
+	}
+}
+
+// Building and Running a k8s manifest using 'podman kube play' command.
+// It will make up to three attempts to run with different configurations,
+// allowing it to run concurrently if necessary
+func BuildAndRunPod() map[string]string {
+	var deployment map[string]string
+	retries := 3
+	for retry := 0; retry < retries; retry++ {
+		deployment := modifyYaml(manifest)
+
+		cmd := exec.Command("podman", "play", "kube", "-")
+		cmd.Stdin = strings.NewReader(deployment["yaml"])
+		output, err := cmd.CombinedOutput()
+		if err == nil {
+			outputString := strings.Split(strings.TrimSpace(string(output)), "\n")
+			containerID := outputString[len(outputString)-1]
+			deployment["containerID"] = containerID
+			return deployment
+		}
+		log.Printf("Failed to start pod. Retrying (attempt %d of %d)...\n", retry+1, retries)
```

Comment:
My suggesting was to have the attempt number reflect the number of the current failed attempt, and this is why I suggested doing a +1 instead of a +2 while droping the word `Retrying`.
```suggestion
		log.Printf("Failed to start pod (attempt %d of %d).\n", retry+1, retries)
```

Reply:
I was pretty sure I removed it, git says otherwise... will fix